lib/core/DocsLayout.js

```diff
@@ -36,15 +36,17 @@ class DocsLayout extends React.Component {
     const metadata = this.props.metadata;
     const content = this.props.children;
     const i18n = translation[this.props.metadata.language];
+    const id = metadata.localized_id;
+    const defaultTitle = metadata.title;
     let DocComponent = Doc;
     if (this.props.Doc) {
       DocComponent = this.props.Doc;
     }
     const title = i18n
-      ? translation[this.props.metadata.language]['localized-strings'][
-          this.props.metadata.localized_id
-        ] || this.props.metadata.title
-      : this.props.metadata.title;
+      ? (i18n['localized-strings'].docs[id] &&
```

Comment:
Seeing this made me realize that we should create a safe getter to access deeply nested object.
We don't want to keep doing this. Example:

```js
// access deeply nested values...
translation &&
translation[this.props.metadata.language] &&
translation[this.props.metadata.language].docs  &&
translation[this.props.metadata.language].docs[id]
```

We can create a small helper function
```js
const idx = (target, path) =>
  path.reduce((obj, key) => (obj && obj[key] ? obj[key] : null), target);
```

So we can write
```js
const title = idx(i18n, ['localized-strings', 'docs', id, 'title']) || defaultTitle;
```

instead of

```js
const title = i18n ? (i18n['localized-strings'].docs[id] && i18n['localized-strings'].docs[id].title) || defaultTitle : defaultTitle;
```

WDYT ?

Reply:
WDYT about using [just-safe-get](https://github.com/angus-c/just/tree/master/packages/object-safe-get)?

Reply:
I used the almost the same method you used to access a deep property, but modified it so that the function accepts the `path` as a string rather than an array, because most of the libraries out there follow the same pattern.

If we need to be more safe then we can move to any library like [just-safe-get](https://github.com/angus-c/just/blob/master/packages/object-safe-get).